taskstatus.go

```diff
@@ -278,10 +288,69 @@ func (tsm *TaskStatusManager) updateStatus(ts TaskStatus, f func(task *TaskRun)
 }
 
 func NewTaskStatusManager(task *TaskRun) *TaskStatusManager {
-	return &TaskStatusManager{
+
+	tsm := &TaskStatusManager{
 		task:                  task,
 		takenUntil:            task.TaskClaimResponse.TakenUntil,
 		status:                task.TaskClaimResponse.Status,
 		statusChangeListeners: map[*TaskStatusChangeListener]bool{},
+		stopReclaiming:        make(chan struct{}),
+		reclaimingDone:        make(chan struct{}),
```

Comment:
If you want, you can make the channel in a variable and assign it to the field... That way you can change the field type to receiving channel... It's just a nits... To prevent other methods from closing the channel...